src/main/java/hudson/scm/PerJobCredentialStore.java

```diff
@@ -105,8 +112,24 @@ private Object writeReplace() {
     private static final ThreadLocal<Boolean> IS_SAVING = new ThreadLocal<Boolean>();
 
     /*package*/ void migrateCredentials(SubversionSCM.DescriptorImpl descriptor) throws IOException {
+        CredentialsStore store = CredentialsProvider.lookupStores(project).iterator().next();
         for (Map.Entry<String, Credential> e : credentials.entrySet()) {
-            descriptor.migrateCredentials(project, e.getKey(), e.getValue());
+            StandardCredentials credential =  descriptor.migrateCredentials(store, e.getKey(), e.getValue());
```

Comment:
Null handling in SVN plugin is quite random and indocumented. It would worth to spend several hours and to document core classes at least (out of the scope of this PR, of course)